src/kOS/Communication/VesselConnection.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Communication;
+using UnityEngine;
+using kOS.Safe.Exceptions;
+using kOS.Safe.Serialization;
+using kOS.Suffixed;
+using TimeSpan = kOS.Suffixed.TimeSpan;
+using kOS.Safe.Communication;
+
+namespace kOS.Communication
+{
+    [kOS.Safe.Utilities.KOSNomenclature("Connection", KOSToCSharp = false)]
+    public class VesselConnection : Connection<kOS.SharedObjects>
+    {
+        private Vessel vessel;
+
+        public override bool Connected {
+            get {
+                return shared.ConnectivityMgr.GetDelay(shared.Vessel, vessel) != -1;
+            }
```

Comment:
Given that the delay is a double and not an int, it's probably good form to either do a ``< 0`` comparison so you're not comparing to a direct number, or to at least use ``-1.0`` or ``-1d`` rather than the integer ``-1``.  My preference is for ``< 0`` rather than just explicitly checking for a single value of ``-1.0``, just because I dislike treating doubles like exact enums, for a variety of reasons.